R/modify_legend.R

```diff
@@ -0,0 +1,191 @@
+
+#' add function call to the overall legend
+#' 
+#' @param object
+#' @param fun.name
+#' @param legend.name
+#' @param \dots
+#' @examples 
+#' gs <- gsplot() %>% 
+#'          points(x=1:5, y=1:5, legend.name = 'points 1') %>% 
+#'          legend()
+#' gs
+#' 
+#' gs <- gsplot() %>% 
+#'          points(x=1:2, y=1:2, col = c("red", "blue"), 
+#'                 legend.name = c('points 1', 'points 2')) %>% 
+#'          legend()
+#' gs
+#' 
+#' @keywords internal
+add_to_legend <- function(object, fun.name, legend.name, ...){
+  
+    if(is.null(legend.name)) {
+      return(object)
```

Comment:
the case where you don't have a legend.name, but you do call legend results in an error:
```r
gs <- gsplot() %>%
   points(1,2) %>%
   legend()
gs
 Error in (function (x, y = NULL, legend, fill = NULL, col = par("col"),  : 
  unused argument (draw = TRUE)
```